test/token-verification.test.js

```diff
@@ -118,97 +148,309 @@ describe('jwt-verification', function() {
+
+          createJWT(payload, DEFAULT_OPTIONS)
+            .then(token => {
+              helpers.assertTokenValidationError(
+                DEFAULT_CONFIG,
+                'asfd',
+                `Not Before time (nbf) claim in the ID token indicates that this token can't be used just yet. Currrent time (${new Date()}) is before ${nbfDate}`,
+                token,
+                done
+              );
+            })
+            .catch(done);
+        });
+
+        it('should validate the exp claim presence', done => {
+          const { expiresIn, ...options } = DEFAULT_OPTIONS;
+
+          createJWT(DEFAULT_PAYLOAD, options)
+            .then(token => {
+              helpers.assertTokenValidationError(
+                DEFAULT_CONFIG,
+                'asfd',
+                'Expiration Time (exp) claim must be a number present in the ID token',
+                token,
+                done
+              );
+            })
+            .catch(done);
         });
+
         it('should validate the token expiration', done => {
           helpers.assertTokenValidationError(
-            {
-              issuer: 'https://wptest.auth0.com/',
-              audience: 'gYSNlU4YC4V1YPdqq8zPQcup6rJw1Mbt'
-            },
+            DEFAULT_CONFIG,
             'asfd',
-            'Expired token.',
-            null,
+            `Expiration Time (exp) claim error in the ID token; current time (${new Date()}) is after expiration time (${defaultExpDate})`,
+            defaultToken,
             done
           );
         });
-        it('should fail with missing claims', done => {
+
+        it('should validate the presence of issued-at claim', done => {
+          const options = Object.assign({}, DEFAULT_OPTIONS, {
+            noTimestamp: true
+          });
+
+          createJWT(DEFAULT_PAYLOAD, options)
+            .then(token => {
+              helpers.assertTokenValidationError(
+                DEFAULT_CONFIG,
+                'asfd',
+                'Issued At (iat) claim must be a number present in the ID token',
+                token,
+                done
+              );
+            })
+            .catch(done);
+        });
+
+        it('should validate the token issued at claim', done => {
+          const iat = Date.now() / 1000 + 20000;
+
+          const payload = Object.assign({}, DEFAULT_PAYLOAD, {
+            iat
+          });
+
+          createJWT(payload, DEFAULT_OPTIONS)
+            .then(token => {
+              helpers.assertTokenValidationError(
+                DEFAULT_CONFIG,
+                'asfd',
+                `Issued At (iat) claim error in the ID token; current time (${new Date()}) is before issued at time (${new Date(
+                  iat * 1000
+                )})`,
+                token,
+                done
+              );
+            })
+            .catch(done);
+        });
+
+        it('should validate presence of auth_time when max_age was specified', done => {
```

Comment:
I'm not seeing any tests for the actual auth_time check either ...

Reply:
Tests have now been added,